sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -269,8 +269,13 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
           }
         }
 
+        def canMerge(joinType: JoinType): Boolean = joinType match {
+          case LeftSingle => false
```

Comment:
in principle, it can do it, but I did not implement it.
as we only need single joins for correlated scalar subqueries with non-equal predicate, the SMJ are unlikely to be useful there anyways.
(for equi-predicates like 
'col = outer(col) .. group by col', we reason that the query returns at most 1 row and don't need the single join).